src/precice/impl/ParticipantImpl.cpp

```diff
@@ -369,7 +378,15 @@ void ParticipantImpl::advance(
 #endif
 
   // Update the coupling scheme time state. Necessary to get correct remainder.
-  const bool   isAtWindowEnd = _couplingScheme->addComputedTime(computedTimeStepSize);
+  const bool isAtWindowEnd = _couplingScheme->addComputedTime(computedTimeStepSize);
+
+  if (_allowsRemeshing && isAtWindowEnd) {
+    int totalMeshesChanges = getTotalMeshChanges();
+    if (reinitHandshake(totalMeshesChanges)) {
+      reinitialize();
+    }
+  }
```

Comment:
We could add an assertion here, right?

```suggestion
  if (_allowsRemeshing) {
    int totalMeshesChanges = getTotalMeshChanges();
    if (isAtWindowEnd){
      if (reinitHandshake(totalMeshesChanges)) {
        reinitialize();
      }
    else {
      assert(totalMeshesChanges == 0);
    }
  }
```

Maybe overkill, but could also help in understanding how things work.